### PR TITLE
Upstream changes on bundle and package tests

### PR DESCRIPTION
**Pull request by J0zi**
Hi team,

this PR consists of all previous PRs (#17, #23) plus fixes from communication with @nmars. All changes are for upstream only and should not affect original pipeline.

Package manifest format:

build upstream catalog
docker install
#quay login
docker install
KIND install
KIND start
kubectl installed
replaced oc with kubectl for upstream
fixed api version for operatorgroup
create namespace
start catalog
addded functionality with waiting on operator to stay alive 1 minute
original parser enabled for upstream
labels added
added info about git clone for catalogue

Bundle format:
can digest package manifest, convert on the fly to bundle and bundle test
ability to test bundle from bundle manifest format
ability to test bundle image

### CODE REVIEW

**Comment by J0zi** [comment]
@nmars, please have a look

**Comment by nmars** [comment]
Searching the output from failed test, the `deploy-olm-operator.yml` playbook failed with the following:

14:21:10  TASK [deploy_olm_operator : fail] **********************************************
14:21:10  task path: /home/jenkins/agent/workspace/cvp-isv-operator-metadata-validation-test/operators/config/ansible/roles/deploy_olm_operator/tasks/main.yml:309
14:21:10  fatal: [localhost]: FAILED! => {"msg": "The conditional check 'operator_uptime.rc != 0' failed. The error was: error while evaluating conditional (operator_uptime.rc != 0): 'dict object' has no attribute 'rc'\n\nThe error appears to be in '/home/jenkins/agent/workspace/cvp-isv-operator-metadata-validation-test/operators/config/ansible/roles/deploy_olm_operator/tasks/main.yml': line 309, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n- fail:\n  ^ here\n"}

**Comment by nmars** [comment]
Sorry, I hit the wrong button by accident and closed the PR! Re-opened now!

**Comment by J0zi** [comment]
@nmars, tests are green. Is there anything we should add? Or is it ready to be merged?

File: roles/build_image_bundle/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    - name: Set input image
+      set_fact:
+        bundle_image: "{{ operator_input_image | regex_replace('[A-z]*://', '') }}"
+        operator_name: "{{ operator_input_image.split(':')[0] | regex_replace('[A-z]*://', '') | basename }}"
+        operator_version: "{{ operator_input_image.split(':')[1] }}"
+    # - fail:
+    #     msg: "{{ operator_name }} {{ operator_version }}"
+    - name: Remove bundle image {{ bundle_image }}
+      shell: "{{ opm_container_tool }} rmi -f {{ bundle_image }}"
+  when: not operator_input_image_convert|bool
```
Comment (dirgim) [inline roles/build_image_bundle/tasks/main.yml:12]: Missing the default value for this, I think it needs to be set to 'false'.

File: roles/build_image_bundle/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+  block:
+    - name: Set input image
+      set_fact:
+        bundle_image: "{{ operator_input_image | regex_replace('[A-z]*://', '') }}"
+        operator_name: "{{ operator_input_image.split(':')[0] | regex_replace('[A-z]*://', '') | basename }}"
+        operator_version: "{{ operator_input_image.split(':')[1] }}"
+    # - fail:
+    #     msg: "{{ operator_name }} {{ operator_version }}"
+    - name: Remove bundle image {{ bundle_image }}
+      shell: "{{ opm_container_tool }} rmi -f {{ bundle_image }}"
```
Comment (dirgim) [inline roles/build_image_bundle/tasks/main.yml:11]: This sets a hard requirement for having docker running, which is not possible in all environments.

File: roles/build_image_bundle/tasks/main.yml (hunk 1 above)
Comment (J0zi) [reply roles/build_image_bundle/tasks/main.yml:12]: solved

File: roles/build_image_bundle/tasks/main.yml (hunk 2 above)
Comment (J0zi) [reply roles/build_image_bundle/tasks/main.yml:11]: set to run only for our stream, upstream kubernetes

File: roles/validate_operator_bundle/tasks/main.yml [hunk 3]
```diff
@@ -16,6 +16,13 @@
   shell: "{{ operator_sdk_bin_path }} bundle validate --verbose {{ operator_work_dir }} > {{ work_dir}}/validation-output.txt 2>&1"
   register: sdk_validation_result
   ignore_errors: true
+  when: not run_upstream|bool
+
+- name: "Validate the operator bundle manifest and metadata with operator-sdk bundle validate from image {{ bundle_image }}"
+  shell: "{{ operator_sdk_bin_path }} bundle validate --verbose {{ bundle_image }} > {{ work_dir}}/validation-output.txt 2>&1"
+  register: sdk_validation_result
```
[the remark this answers is not in the document]
Comment (mvalahtv) [reply roles/validate_operator_bundle/tasks/main.yml:23]: It is strange. Could you try to comment `run_upstream==true` part and try again.

[the remark this answers is not in the document]
Comment (J0zi) [reply roles/validate_operator_bundle/tasks/main.yml:17]: @dirgim, I changed order, hope it will help.

[the remark this answers is not in the document]
Comment (J0zi) [reply roles/validate_operator_bundle/tasks/main.yml:20]: issue solved

**Comment by J0zi** [comment]
@dirgim, will be great to label PR by `/ok-to-test`, so I can rerun tests:)

**Comment by dirgim** [comment]
@J0zi triggering works for the CVP CI anyways since it doesn't use prow so the warning message can be ignored - even with that label it will complain, but I added it to look nicer :)

**Comment by J0zi** [comment]
@dirgim, I don't see test results in detail, so cannot fix anything. Please provide error messages.

**Comment by J0zi** [comment]
@dirgim, any update?

**Comment by dirgim** [comment]
@J0zi it seems like it's an error in testing the CVP side, we're working on fixing it

**Comment by J0zi** [comment]
@dirgim, any update for us? :)

**Comment by J0zi** [comment]
@dirgim, you can continue. Duplicated deploy role to separate one for upstream Kubernetes.

**Comment by dirgim** [comment]
@nmars @lslebodn @samvarankashyap @scoheb @sonam1412 can you take a look?

File: roles/build_image_bundle/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+          stat:
+            path: "{{ annotation_file_path }}"
+          register: annotation_file_st
+
+        - name: Fail if annotation file doesn't exists
+          fail:
+          when: not annotation_file_st.stat.exists
+
+        - name: "Getting content of annotation.yaml file"
+          shell: "cat {{ annotation_file_path }}"
```
[the remark this answers is not in the document]
Comment (mvalahtv) [reply roles/build_image_bundle/tasks/main.yml:74]: @samvarankashyap you cannot `include_vars` when yaml file is on remote host. That was reason we did it that way

**Comment by J0zi** [comment]
@samvarankashyap, thanks for the review. Updated, please check.

**Comment by J0zi** [comment]
@samvarankashyap @dirgim, is there something we should change or it can be merged?

**Comment by J0zi** [comment]
@nmars @samvarankashyap, any update?

**Comment by J0zi** [comment]
@nmars @samvarankashyap, please specify ETA for this merge. Is there anything missing?

**Comment by nmars** [comment]
Sorry for the delay. We need to verify the results of some supplemental testing due to the broad scope of the changes, and will have something more definitive tomorrow.

**Comment by nmars** [comment]
lgtm

**Pull request merged**